superset/queries/dao.py

```diff
@@ -48,3 +50,20 @@ def update_saved_query_exec_info(query_id: int) -> None:
                 saved_query.rows = query.rows
                 saved_query.last_run = datetime.now()
             db.session.commit()
+
+    @staticmethod
+    def save_metadata(query: Query, query_payload: str) -> None:
+        # parse payload
+        try:
+            payload: Dict[str, Any] = json.loads(query_payload)
```

Comment:
The fact that we are dumping a json then reloading it within the same process doesn't bod well for me. It's probably also not a great idea to passing strings around between functions.

Can we refactor `_execution_context_convertor` to return the raw dict instead? (Maybe add a new public function?)

Reply:
@ktmud so looking at the code we'd have to do a `json.dump` before returning the payload for it to fit into the `json_success` function which needs a `str`

https://github.com/hve-labs/superset/blob/060b5c0e177b89eff9b6ace83b7df235f25cc198/superset/views/base.py#L159

Reply:
Yeah, but can we maybe update [`ExecutionContextConvertor.to_payload`](https://github.com/apache/superset/commit/0d0c759cfed3dc72aaca029b5da1db1acfa393a9) to return Python dict and only do JSON dumps in the [`ExecuteSqlCommand.run`](https://github.com/apache/superset/blob/8e29ec5a6685867ffc035d20999c54c2abe36fb1/superset/sqllab/command.py#L99-L101)? Looks like `to_payload` is only used in this one place.